Mage/src/main/java/mage/abilities/keyword/PlotAbility.java

```diff
@@ -30,6 +50,271 @@ public PlotAbility copy() {
+            game.informPlayers(owner.getLogName() + " plots " + mainCard.getLogName());
+            game.fireEvent(GameEvent.getEvent(GameEvent.EventType.BECOME_PLOTTED, mainCard.getId(), source, owner.getId()));
+        }
+        return true;
+    }
+}
+
+/**
+ * Exile the source card in the plot exile zone of its owner
+ * and allow its owner to cast it at sorcery speed starting
+ * next turn.
+ */
+class PlotSourceExileEffect extends OneShotEffect {
+
+    PlotSourceExileEffect() {
+        super(Outcome.Benefit);
+    }
+
+    private PlotSourceExileEffect(final PlotSourceExileEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public PlotSourceExileEffect copy() {
+        return new PlotSourceExileEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return PlotAbility.doExileAndPlotCard(game.getCard(source.getSourceId()), game, source);
+    }
+}
+
+class PlotAddSpellAbilityEffect extends ContinuousEffectImpl {
+
+    private final MageObjectReference mor;
+
+    PlotAddSpellAbilityEffect(MageObjectReference mor) {
+        super(Duration.EndOfGame, Layer.AbilityAddingRemovingEffects_6, SubLayer.NA, Outcome.AddAbility);
+        this.mor = mor;
+        staticText = "Plot card";
+    }
+
+    private PlotAddSpellAbilityEffect(final PlotAddSpellAbilityEffect effect) {
+        super(effect);
+        this.mor = effect.mor;
+    }
+
+    @Override
+    public PlotAddSpellAbilityEffect copy() {
+        return new PlotAddSpellAbilityEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Card card = mor.getCard(game);
+        if (card == null) {
+            discard();
+            return true;
+        }
+
+        UUID mainCardId = card.getMainCard().getId();
+        Player player = game.getPlayer(card.getOwnerId());
+        if (game.getState().getZone(mainCardId) != Zone.EXILED || player == null) {
+            discard();
+            return true;
+        }
+
+        List<Card> faces = CardUtil.getCastableComponents(card, null, source, player, game, null, false);
+        for (Card face : faces) {
+            // Add the spell ability to each castable face.
+            PlotSpellAbility ability = new PlotSpellAbility();
+            ability.setSourceId(face.getId());
+            ability.setControllerId(player.getId());
+            ability.setSpellAbilityType(face.getSpellAbility().getSpellAbilityType());
+            ability.setAbilityName(face.getName());
+            game.getState().addOtherAbility(face, ability);
+        }
+        return true;
     }
 }
+
+class PlotSpellAbility extends SpellAbility {
+
+    private String abilityName;
+    private SpellAbility spellAbilityToResolve;
+
+    PlotSpellAbility() {
+        super(null, "Plot", Zone.EXILED, SpellAbilityType.BASE_ALTERNATE, SpellAbilityCastMode.NORMAL);
+        this.setAdditionalCostsRuleVisible(false);
+        this.name = "Cast with Plot";
```

Comment:
I was using the Foretell method as a reference, which was a bit confused on those parts.
I did tweak that part in the code since then, my intent was:

-> Display all castable modes specifying that those were plotted (for 'without paying mana cost, at sorcery speed')
![image](https://github.com/magefree/mage/assets/34709007/614b952a-aa09-4ae9-ac9e-5f62df88d369)
-> Have the tests identify that we want to cast the card from exiled, as it is plotted.

I'll look into setSpellName if that accomplishes what I want.

Reply:
Must use standard name notation like “Plot card name” (it used in suspend/foretell).

Reply:
Alright I don't think Foretell is set up correctly, but a foretold card should have an alternative cost to be cast.
We don't have the comprehensive rules updated yet for Plot, but I would assume a plotted card  can be cast with a no mana cost alternative cost.

As such, let's display the cast action like this (and as demonstrated in the screenshot, the Morph alternative cost can not be used there): 
![image](https://github.com/magefree/mage/assets/34709007/8d9d5fa6-2811-454a-8598-70a53b470c93)

With the "using Plot" suffix, the test suite (and real game too) is able to distinguish the cast using the plotted alternative cost from the other alternative costs that may apply:
![image](https://github.com/magefree/mage/assets/34709007/ea970534-e2ce-4977-b3e5-499c1871b2da)

Reply:
The relevant Foretell ruling
> 702.143a. [Foretell](https://yawgatog.com/resources/magic-rules/#foretell) is a keyword that functions while the [card](https://yawgatog.com/resources/magic-rules/#card) with [foretell](https://yawgatog.com/resources/magic-rules/#foretell) is in a [player](https://yawgatog.com/resources/magic-rules/#player)'s [hand](https://yawgatog.com/resources/magic-rules/#hand). Any time a [player](https://yawgatog.com/resources/magic-rules/#player) has [priority](https://yawgatog.com/resources/magic-rules/#priority) [during](https://yawgatog.com/resources/magic-rules/#during) their turn, that [player](https://yawgatog.com/resources/magic-rules/#player) may [pay](https://yawgatog.com/resources/magic-rules/#pay) {2} and [exile](https://yawgatog.com/resources/magic-rules/#exile) a [card](https://yawgatog.com/resources/magic-rules/#card) with [foretell](https://yawgatog.com/resources/magic-rules/#foretell) from their [hand](https://yawgatog.com/resources/magic-rules/#hand) [face down](https://yawgatog.com/resources/magic-rules/#face_down). That [player](https://yawgatog.com/resources/magic-rules/#player) may look at that [card](https://yawgatog.com/resources/magic-rules/#card) as long as it remains in [exile](https://yawgatog.com/resources/magic-rules/#exile). They may [cast](https://yawgatog.com/resources/magic-rules/#cast) that [card](https://yawgatog.com/resources/magic-rules/#card) after the current turn has ended by paying any [foretell](https://yawgatog.com/resources/magic-rules/#foretell) [cost](https://yawgatog.com/resources/magic-rules/#cost) it has rather than paying that [spell](https://yawgatog.com/resources/magic-rules/#spell)'s [mana cost](https://yawgatog.com/resources/magic-rules/#mana_cost). Casting a [spell](https://yawgatog.com/resources/magic-rules/#spell) this way follows the rules for paying [alternative costs](https://yawgatog.com/resources/magic-rules/#alternative_cost) in rules [601.2b](https://yawgatog.com/resources/magic-rules/#R6012b) and [601.2f](https://yawgatog.com/resources/magic-rules/#R6012f)-h.